Update Python SDK to use API v1

banool opened the pull request:
## Description
This PR updates our Python SDK to work with the v1 API.

While I'm here I add package management and command runner stuff and update the README.

CI next to come.

## Test Plan
```
make fmt
make examples
make test
```

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/aptos-labs/aptos-core/2600)
<!-- Reviewable:end -->

gregnazario commented on ecosystem/python/sdk/setup.py:
```diff
@@ -19,5 +19,5 @@
     packages=["aptos_sdk"],
     python_requires='>=3.7',
     url="https://github.com/aptos-labs/aptos-core",
-    version="0.1.0.0",
```
lol lots of 0s

davidiw replied:
No need as it's not been published

gregnazario commented on ecosystem/python/sdk/examples/token.py:
```diff
@@ -19,8 +20,8 @@
     print(f"Bob: {bob.address()}")
 
-    faucet_client.fund_account(alice.address(), 20_000)
-    faucet_client.fund_account(bob.address(), 20_000)
+    faucet_client.fund_account(alice.address(), 20_000_000_000)
+    faucet_client.fund_account(bob.address(), 20_000_000_000)
```
Uh, let's not change this that high.

gregnazario commented on ecosystem/python/sdk/examples/coin.py:
```diff
@@ -16,7 +17,7 @@
     print(f"Bob: {bob.address()}")
 
-    faucet_client.fund_account(alice.address(), 20_000)
+    faucet_client.fund_account(alice.address(), 20_000_000_000)
```
similarly

gregnazario commented on ecosystem/python/sdk/README.md:
````diff
@@ -1,47 +1,50 @@
+```
 
-To run all unit tests:
+Next, tell the E2E tests to talk to this locally running testnet:
````
Let's spell out end to end.

gregnazario commented on ecosystem/python/sdk/aptos_sdk/account.py:
```diff
@@ -7,19 +7,17 @@
 
-from . import ed25519
 from .account_address import AccountAddress
+from .ed25519 import PrivateKey, PublicKey
```
It's funny cause I think this was changed the other way recently

davidiw replied:
I prefer the other way for the key imports. It was intentional for if and when we sort other key types.

banool replied:
I don't mind, I'll change it back.

gregnazario commented on ecosystem/python/sdk/aptos_sdk/client.py:
```diff
@@ -117,21 +118,21 @@ def submit_transaction(self, sender: Account, payload: Dict[str, Any]) -> str:
         txn_request = {
             "sender": f"{sender.address()}",
             "sequence_number": str(self.account_sequence_number(sender.address())),
-            "max_gas_amount": "2000",
+            "max_gas_amount": str(1_000_000),
```
similar

gregnazario reviewed (approved):
Let's turn the example amounts down to the amounts that they should be, otherwise looks good

davidiw reviewed (commented):
overall its a great pr

* ed25519 import was intentional, let's roll back the change, please
* I think some config on your computer has set the line length longer than the default, please double check, because the auto-formatting seems
* ideally the API for function takes in a a single string otherwise let's fix upstream -- that is unintuitive -- the json object is unintuitive
* I've noticed that more often than not python is not autolinked to python3, why the change?

I'd be really happy with the most trivial of PRs to update it and in fact, I already borrowed quite a bit of it to help push out an update to the SDK.

I want to better understand the requirements around make and poetry and their general accessibility. For example, on windows, I can easily install python without a unix environment. Make is only available for unix environments.

Of course one could argue that xargs isn't particularly cross platform :dagger:

banool commented:
Addressed the feedback about the ed25519 module, the coin amounts, and the string representation of move types discussed elsewhere.

As for poetry, I see that it is supported on both Unix systems and Windows.
